Source/WebCore/html/HTMLTextAreaElement.h

```diff
@@ -119,28 +96,46 @@ class HTMLTextAreaElement final : public HTMLTextFormControlElement {
     RenderPtr<RenderElement> createElementRenderer(RenderStyle&&, const RenderTreePosition&) final;
     bool appendFormData(DOMFormData&) final;
     void reset() final;
-    bool hasCustomFocusLogic() const final;
-    int defaultTabIndex() const final;
-    bool isMouseFocusable() const final;
-    bool isKeyboardFocusable(KeyboardEvent*) const final;
+    bool hasCustomFocusLogic() const final { return true; }
+    int defaultTabIndex() const final { return 0; }
+    bool isMouseFocusable() const final { return isFocusable(); }
+    bool isKeyboardFocusable(KeyboardEvent*) const final { return isFocusable(); }
     void updateFocusAppearance(SelectionRestorationMode, SelectionRevealMode) final;
 
-    bool accessKeyAction(bool sendMouseEvents) final;
+    bool accessKeyAction(bool) final { focus(); return false; }
 
-    bool shouldUseInputMethod() final;
-    bool matchesReadWritePseudoClass() const final;
+    bool shouldUseInputMethod() final { return true; }
+    bool matchesReadWritePseudoClass() const final { return isMutable(); }
 
-    bool valueMissing(const String& value) const { return isRequiredFormControl() && isMutable() && value.isEmpty(); }
-    bool tooShort(StringView, NeedsToCheckDirtyFlag) const;
-    bool tooLong(StringView, NeedsToCheckDirtyFlag) const;
+    bool valueMissing() const final;
+    bool tooShort() const final;
+    bool tooLong() const final;
+    bool isValidValue(StringView) const;
 
-    unsigned m_rows;
-    unsigned m_cols;
-    WrapMethod m_wrap { SoftWrap };
+    bool valueMissing(StringView valueOverride) const;
+    bool tooShort(StringView valueOverride, NeedsToCheckDirtyFlag) const;
+    bool tooLong(StringView valueOverride, NeedsToCheckDirtyFlag) const;
+
+    RefPtr<TextControlInnerTextElement> innerTextElementCreatingShadowSubtreeIfNeeded() final;
+    RenderStyle createInnerTextStyle(const RenderStyle&) final;
+    void copyNonAttributePropertiesFromElement(const Element&) final;
+
+    bool willRespondToMouseClickEventsWithEditability(Editability) const final { return !isDisabledFormControl(); }
+
+    RenderTextControlMultiLine* renderer() const;
+
+    enum WrapMethod : uint8_t { NoWrap, SoftWrap, HardWrap };
+
+    static constexpr int defaultRows = 2;
+    static constexpr int defaultCols = 20;
+
+    unsigned m_rows { defaultRows };
+    unsigned m_cols { defaultCols };
     RefPtr<HTMLElement> m_placeholder;
     mutable String m_value;
+    WrapMethod m_wrap { SoftWrap };
     mutable bool m_isDirty { false };
```

Comment:
We should probably change these to uint8_t so that they're tightly packed on Windows.